File: TiaCodegen/Blocks/Network.cs
using TiaCodegen.Commands;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Blocks
{
    public class Network : BaseOperationOrSignal
    {
        public string NetworkTitle { get; set; }

        public string Description { get; set; }

        public string NetworkTitleEnglish { get; set; }

        public string DescriptionEnglish { get; set; }

        public Network(params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        { }


        public Network(string networkTitle, string networkTitleEnglish, params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        {
            NetworkTitle = networkTitle;
            NetworkTitleEnglish = networkTitleEnglish;
        }
    }
}

File: TiaCodegen-ts/src/Blocks/Network.ts
import { IOperationOrSignal } from '../Interfaces/IOperationOrSignal';
import { BaseOperationOrSignal } from '../Commands/BaseOperationOrSignal';

export class Network extends BaseOperationOrSignal {
    networkTitle: string | undefined;
    description: string | undefined;
    networkTitleEnglish: string | undefined;
    descriptionEnglish: string | undefined;

    constructor(
        networkTitleOrOp?: string | IOperationOrSignal,
        networkTitleEnglishOrOp?: string | IOperationOrSignal,
        ...operationOrSignals: IOperationOrSignal[]
    ) {
        super();
        if (typeof networkTitleOrOp === 'string') {
            this.networkTitle = networkTitleOrOp;
            this.networkTitleEnglish = typeof networkTitleEnglishOrOp === 'string'
                ? networkTitleEnglishOrOp
                : undefined;
            if (networkTitleEnglishOrOp !== undefined && typeof networkTitleEnglishOrOp !== 'string') {
                this.children.push(networkTitleEnglishOrOp);
            }
            for (const op of operationOrSignals) {
                this.children.push(op);
            }
        } else {
            if (networkTitleOrOp !== undefined) this.children.push(networkTitleOrOp);
            if (networkTitleEnglishOrOp !== undefined && typeof networkTitleEnglishOrOp !== 'string') {
                this.children.push(networkTitleEnglishOrOp);
            }
            for (const op of operationOrSignals) {
                this.children.push(op);
            }
        }
    }
}
